support sharing_nfs in TrueNAS Community Edition 25.04

devietti opened the pull request:
A small change I had to make to get NFS exports to work properly in 25.04, which has type COMMUNITY_EDITION instead of SCALE.

Previously, due to the change in "version type", the older NFS1 code would be selected, resulting in this error message:
```
Error creating NFS export \"None\": midclt exited with status 1: \"[EINVAL] data.path: Field required\n[EINVAL] data.comment: Input should be a valid string\n[EINVAL] data.paths: Extra inputs are not permitted\n\"
```
Now, TrueNAS version 25.04 uses `nfs2()` instead which works great.

devietti committed:
SCALE rebranded as COMMUNITY_EDITION in early 2025

MozzieBytes reviewed (commented):
Added a suggestion. 

Good catch on this implementation, I completely missed the change to Community Edition when looking into this

devietti committed:
cleaner checking of TrueNAS system type

Also support Enterprise installations. These seem like the right strings per https://api.truenas.com/v25.04.2/api_methods_system.product_type.html

MozzieBytes commented:
@arensb hope you're doing well.

What would be the next steps to get this PR reviewed and merged in?

iX Systems have sunset the TrueNAS Scale name from the 2025 versions. This means management of NFS shares is currently broken upstream without this change :)

arensb commented:
Sorry, haven't had much time to devote to this project. I'll try to find some.

arensb commented:
This change (using a set rather than a list) was just made in 18627fb0c5bbc7f3f4a13857283fb333b3dc8bba.
